src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_attend_and_excite.py

```diff
@@ -0,0 +1,964 @@
+        print(f"\t Finished with loss of: {loss}")
+        return loss, latents, max_attention_per_index
+
+    def register_attention_control(self):
+        attn_procs = {}
+        cross_att_count = 0
+        for name in self.unet.attn_processors.keys():
+            if name.startswith("mid_block"):
+                place_in_unet = "mid"
+            elif name.startswith("up_blocks"):
+                place_in_unet = "up"
+            elif name.startswith("down_blocks"):
+                place_in_unet = "down"
+            else:
+                continue
+
+            cross_att_count += 1
+            attn_procs[name] = AttendExciteCrossAttnProcessor(
+                attnstore=self.attention_store, place_in_unet=place_in_unet
+            )
+
+        self.unet.set_attn_processor(attn_procs)
+        self.attention_store.num_att_layers = cross_att_count
+
+    def show_cross_attention(
+        self,
+        prompt: str,
+        indices_to_alter: List[int],
+        res: int,
+        from_where: List[str],
+        select: int = 0,
+        orig_image=None,
+    ):
+        tokens = self.tokenizer.encode(prompt)
+        decoder = self.tokenizer.decode
+        attention_maps = self.attention_store.aggregate_attention(res, from_where, True, select).detach().cpu()
+        images = []
+
+        # show spatial attention for indices of tokens to strengthen
+        for i in range(len(tokens)):
+            image = attention_maps[:, :, i]
+            if i in indices_to_alter:
+                image = show_image_relevance(image, orig_image)
+                image = image.astype(np.uint8)
+                image = np.array(Image.fromarray(image).resize((res**2, res**2)))
+                image = text_under_image(image, decoder(int(tokens[i])))
+                images.append(image)
+
+        return view_images(np.stack(images, axis=0))
+
+    @torch.no_grad()
+    def __call__(
+        self,
+        prompt: Union[str, List[str]],
+        indices_to_alter: List[int],
+        attention_res: int = 16,
+        height: Optional[int] = None,
+        width: Optional[int] = None,
+        num_inference_steps: int = 50,
+        guidance_scale: float = 7.5,
+        negative_prompt: Optional[Union[str, List[str]]] = None,
+        num_images_per_prompt: Optional[int] = 1,
+        eta: float = 0.0,
+        generator: Optional[Union[torch.Generator, List[torch.Generator]]] = None,
+        latents: Optional[torch.FloatTensor] = None,
+        prompt_embeds: Optional[torch.FloatTensor] = None,
+        negative_prompt_embeds: Optional[torch.FloatTensor] = None,
+        output_type: Optional[str] = "pil",
+        return_dict: bool = True,
+        callback: Optional[Callable[[int, int, torch.FloatTensor], None]] = None,
+        callback_steps: Optional[int] = 1,
+        cross_attention_kwargs: Optional[Dict[str, Any]] = None,
+        max_iter_to_alter: Optional[int] = 25,
+        run_standard_sd: bool = False,
```

Comment:
what does this do exactly?

Reply:
if `run_standard_sd == False`, it will run stable diffusion without Attend and Excite. It's nice that you can generate comparison images, but maybe not really needed here?